### PR TITLE
BT/OT l2 switch to new net_pkt api

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Review by rveerama1 — APPROVED** [review]
BT changes LGTM

File: subsys/net/l2/openthread/openthread.c [hunk 1]
```diff
@@ -108,45 +108,37 @@ void ot_receive_handler(otMessage *aMessage, void *context)
 	u16_t offset = 0U;
 	u16_t read_len;
 	struct net_pkt *pkt;
-	struct net_buf *prev_buf = NULL;
+	struct net_buf *pkt_buf;
 
-	pkt = net_pkt_get_reserve_rx(K_NO_WAIT);
+	pkt = net_pkt_rx_alloc_with_buffer(ot_context->iface,
+					   otMessageGetLength(amessage),
```
Comment (rlubos) [inline subsys/net/l2/openthread/openthread.c]: It's `aMessage` - compilation error.

**Review by rlubos — CHANGES_REQUESTED** [review]

**Comment by tbursztyka** [comment]
hum CI does not built OT anymore as well.

**Comment by rlubos** [comment]
With OpenThread it's a little trickier - there were complaints that OT builds are too slow -> #12206, as it currently downloads OT repo during build. Hopefully, once west allows us to integrate OT repository, we could reenable them.

**Comment by tbursztyka** [comment]
Ok

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
Tested both with nRF52840, seem to work fine 👍

**Pull request merged**